ee/frontend/mobile-replay/mobile.types.ts

```diff
@@ -0,0 +1,155 @@
+// copied from rrweb-snapshot, not included in rrweb types
+import { customEvent, EventType } from '@rrweb/types'
+
+export enum NodeType {
+    Document = 0,
+    DocumentType = 1,
+    Element = 2,
+    Text = 3,
+    CDATA = 4,
+    Comment = 5,
+}
+
+export type documentNode = {
+    type: NodeType.Document
+    childNodes: serializedNodeWithId[]
+    compatMode?: string
+}
+
+export type documentTypeNode = {
+    type: NodeType.DocumentType
+    name: string
+    publicId: string
+    systemId: string
+}
+
+export type attributes = {
+    [key: string]: string | number | true | null
+}
+
+export type elementNode = {
+    type: NodeType.Element
+    tagName: string
+    attributes: attributes
+    childNodes: serializedNodeWithId[]
+    isSVG?: true
+    needBlock?: boolean
+    // This is a custom element or not.
+    isCustom?: true
+}
+
+export type textNode = {
+    type: NodeType.Text
+    textContent: string
+    isStyle?: true
+}
+
+export type cdataNode = {
+    type: NodeType.CDATA
+    textContent: ''
+}
+
+export type commentNode = {
+    type: NodeType.Comment
+    textContent: string
+}
+
+export type serializedNode = (documentNode | documentTypeNode | elementNode | textNode | cdataNode | commentNode) & {
+    rootId?: number
+    isShadowHost?: boolean
+    isShadow?: boolean
+}
+
+export type serializedNodeWithId = serializedNode & { id: number }
+
+// end copied section
+
+export type MobileNodeType = 'text' | 'image' | 'rectangle' | 'div'
+
+export type MobileStyles = {
+    color?: string
+    backgroundColor?: string
+    /**
+     * @description if borderWidth is present, then border style is assumed to be solid
+     */
+    borderWidth?: string | number
+    /**
+     * @description if borderRadius is present, then border style is assumed to be solid
+     */
+    borderRadius?: string | number
+    /**
+     * @description if borderColor is present, then border style is assumed to be solid
+     */
+    borderColor?: string
+}
+
+type wireframeBase = {
+    id: number
+    /**
+     * @description x and y are the top left corner of the element, if they are present then the element is absolutely positioned
+     */
+    x: number
+    y: number
+    width: number
+    height: number
+    childWireframes?: wireframe[]
+    type: MobileNodeType
+    style?: MobileStyles
+}
+
+export type wireframeText = wireframeBase & {
+    type: 'text'
+    text: string
+}
+
+export type wireframeImage = wireframeBase & {
+    type: 'image'
+    /**
+     * @description this will be used as base64 encoded image source, with no other attributes it is assumed to be a PNG
+     */
+    base64: string
+}
+
+export type wireframeRectangle = wireframeBase & {
+    type: 'rectangle'
+}
+
+export type wireframeDiv = wireframeBase & {
+    /*
+     * @description this is the default type, if no type is specified then it is assumed to be a div
+     */
+    type: 'div'
+}
```

Comment:
What's the diff between the rectangle and the div?

Reply:
rectangle is implemented as an SVG, it was only not implemented as a div because if we want e.g. circle that'll be easier as svg so I thought we may as well start somewhere

div is a div :) So mobile doesn't ever need to send it (but could). I'll use it as a fallback when I need structure